java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java

```diff
@@ -140,12 +140,16 @@ public StorageEventRequest buildRequest(String sourceBucketName,
                 .map(inputBasePath -> sourceObjectPath.replace(inputBasePath, ""))
                 .orElse(sourceObjectPath);
 
+        boolean compressedInput = compressedSource(sourceObjectPath, sourceContentEncoding);
 
         StorageEventRequest request = StorageEventRequest.builder()
             .sourceBucketName(sourceBucketName)
             .sourceObjectPath(sourceObjectPath)
             .destinationBucketName(transform.getDestinationBucketName())
             .destinationObjectPath(transform.getPathWithinBucket() + sourceObjectPathWithinBase)
+            .decompressInput(compressedInput)
+            //TODO: specify compressOutput as part of the rule. For now, follow the input
+            .compressOutput(compressedInput)
```

Comment:
Missing parameters, so treated input as uncompressed always

Reply:
i prefer `inputIsCompressed` or `isCompressedInput`. `compressedInput` sounds like it's a compressed stream of the input or something.